### PR TITLE
Pip authenticate task v0

### PR DESCRIPTION
**Pull request by shubham90**
This task provides authentication for pip client used for installing python distributions. It sets an environment variable which is shared by all the tasks.

### CODE REVIEW

File: Tasks/PipAuthenticateV0/Authentication.ts [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+        switch(scheme) {
+            case "token":
+                const token = externalAuth.parameters["apitoken"];
+                tl.debug(tl.loc("Info_AddingTokenAuthEntry", feedUri));
+                externalAuthArray.push(new ExternalAuthInfo(<IPackageSource>
+                    {
+                        feedUri: feedUri
+                    },
+                    ExternalAuthType.Token,
+                    "azDev",
```
Comment (jmyersmsft) [inline Tasks/PipAuthenticateV0/Authentication.ts]: Same "AzDev" comment as the other PR

File: Tasks/PipAuthenticateV0/PipAuthenticateMain.ts [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+}
+
+// Telemetry
+function _logUniversalStartupVariables() {
+    try {
+        const pipAuthenticateTelemetry = {
+            "buildProperties": tl.getInput("buildProperties"),
+            "basePath": tl.getInput("basePath"),
+            "System.TeamFoundationCollectionUri": tl.getVariable("System.TeamFoundationCollectionUri"),
+            "solution": tl.getInput("solution"),
```
Comment (jmyersmsft) [inline Tasks/PipAuthenticateV0/PipAuthenticateMain.ts]: Looks like some copypasta: these `getInput` calls are referencing inputs that aren't declared on this task. Same on the Twine PR.

Comment (shubham90) [reply Tasks/PipAuthenticateV0/PipAuthenticateMain.ts]: Yes, good catch. It was an oversight on my part. Will change.

File: Tasks/PipAuthenticateV0/Authentication.ts [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+    if (!endpointNames || endpointNames.length === 0)
+    {
+        return externalAuthArray;
+    }
+
+    tl.debug(tl.loc("Info_AddingExternalFeeds", endpointNames.length));
+    endpointNames.forEach((endpointName: string) => {
+        let feedUri = tl.getEndpointUrl(endpointName, false);
+        utils.formPipCompatibleUri("", "", feedUri);
```
Comment (salvmd) [inline Tasks/PipAuthenticateV0/Authentication.ts]: remove

**Pull request merged**